src/os/bluestore/BitmapFreelistManager.cc

```diff
@@ -113,33 +113,32 @@ int BitmapFreelistManager::init()
 
   // load meta
   while (it->valid()) {
-    string k = it->key();
-    if (k == "bytes_per_block") {
+    if (it->key() == "bytes_per_block") {
```

Comment:
key() returns a string by value, not reference, so c_str() here is a use-after-free